core/modules/node/node.module

```diff
@@ -1470,8 +1470,15 @@ function node_search_execute($keys = NULL, $conditions = NULL) {
 /**
  * Implements hook_ranking().
  */
-function node_ranking() {
+function node_ranking($keywords = NULL) {
   // Create the ranking array and add the basic ranking options.
+  $words = explode(' ', $keywords);
+  $conditions = array();
+  foreach ($words as $word) {
+    $conditions[] = "n.title LIKE '%" . $word . "%'";
```

Comment:
This looks like a prime candidate for SQL-injection. I'm not sure we can safely use user-entered values directly in a SQL query like this. We probably could however, pass placeholders like `array(':keyword1', ':keyword2', ':keyword3')`, and let the rest of the search query populate those placeholders.

Reply:
Actually on further thought yet, that wouldn't work because parameters have to be wrapped in `db_like()` if being passed into a LIKE statement like this. Further compounding this problem this approach would not respect search's querying structure, such as doing `term1 AND term2` vs `term1 OR term2`. The AND/OR would end up being part of the query, rather than an operator as designed.

Overall I think we might be chasing the wrong approach here. Simply giving the H1 a greater weight in the index or adding the title in a different way to `_node_index_node()` might accomplish this goal just as effectively.